systems/registry/node/pkg/db/nodestatus_repo.go

```diff
@@ -0,0 +1,92 @@
+package db
+
+import (
+	log "github.com/sirupsen/logrus"
+	"github.com/ukama/ukama/systems/common/sql"
+	"github.com/ukama/ukama/systems/common/ukama"
+	"gorm.io/gorm"
+)
+
+type NodeStatusRepo interface {
+	Update(*NodeStatus) error
+	Get(ukama.NodeID) (*NodeStatus, error)
+	Delete(ukama.NodeID) error
+	GetAll() ([]NodeStatus, error)
+	GetNodeCount() (activeNodeCount, inactiveNodeCount int64, err error)
+}
+
+type nodeStatusRepo struct {
+	Db sql.Db
+}
+
+func NewNodeStatusRepo(db sql.Db) NodeStatusRepo {
+	return &nodeStatusRepo{
+		Db: db,
+	}
+}
+
+func (n *nodeStatusRepo) Update(ns *NodeStatus) error {
+	err := n.Db.GetGormDb().Transaction(func(tx *gorm.DB) error {
+		t := tx.Where("node_id= ?", ns.NodeId).Delete(&NodeStatus{})
```

Comment:
Make sure you send lowercased nodeIds against the DB with ns.NodeId.StringLowercase().